File: 0410-split-array-largest-sum/0410-split-array-largest-sum.java
class Solution {
    public int splitArray(int[] nums, int k) {
        int min=0;
        int max=0;
        for(int i:nums){
            min=Math.max(i,min);
            max+=i;
        }
        int ans=-1;
        while(min<=max){
            int mid=(min+max)/2;
            if(isPos(mid,nums,k)){
                ans=mid;
                max=mid-1;

            }else{
                
                min=mid+1;
            }
        }
        return ans;
    }
    boolean isPos(int m,int a[],int k){
        int s=1;
        int c=0;
        for(int i:a){
            if((c+i)>m){
                s++;
                c=i;
            }else{
                c+=i;
            }
        }
        return s<=k;
    }
}

File: 0410-split-array-largest-sum/0410-split-array-largest-sum.py
class Solution:
    def splitArray(self, nums: List[int], k: int) -> int:
        l=max(nums)
        r=sum(nums)
        ans=-1
        while(l<=r):
            m=l+(r-l)//2
            if(self.isPos(nums,m,k)):
                ans=m
                r=m-1
            else:
                l=m+1
        return ans
    def isPos(self,nums:List[int],m:int,k:int)->bool:
        t=1
        sum=0
        for i in nums:
            sum+=i
            if(sum>m):
                t+=1
                sum=i
        return t<=k

        